extensions/amp-empower/0.1/amp-empower.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {addParamsToUrl} from '../../../src/url';
+import {devAssert, userAssert} from '../../../src/log';
+import {dict} from '../../../src/utils/object';
+import {getDataParamsFromAttributes} from '../../../src/dom';
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+class AmpEmpower extends AMP.BaseElement {
+  /**
+   *Creates an instance of AmpEmpower.
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element);
+    /** @private {?string}  */
+    this.video_ = null;
+    /** @private {?Element} */
+    this.iframe_ = null;
+    /** @private {?string} */
+    this.videoIframeSrc_ = null;
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url(this.getVideoIframeSrc_());
+    this.preconnect.url('https://cdn.empower.net', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.video_ = userAssert(
+      this.element.getAttribute('data-video'),
+      'The data-video attribute is required for <amp-empower> %s',
+      this.element
+    );
+  }
+
+  /** @return {string} */
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+
+    devAssert(this.video_);
+    let src =
```

Comment:
nit: Can we make this a `const baseSrc`, and then assign inline?

```js
return (this.videoIframeSrc_ = addParamsToUrl(baseSrc, params));
```

Reply:
Done 👍